django/urls/resolvers.py

```diff
@@ -478,11 +478,19 @@ def _reverse_with_prefix(self, lookup_view, _prefix, *args, **kwargs):
             lookup_view_s = lookup_view
 
         patterns = [pattern for (possibility, pattern, defaults) in possibilities]
-        raise NoReverseMatch(
-            "Reverse for '%s' with arguments '%s' and keyword "
-            "arguments '%s' not found. %d pattern(s) tried: %s" %
-            (lookup_view_s, args, kwargs, len(patterns), patterns)
-        )
+        if patterns:
+            if args:
+                msg = ("Reverse for '%s' with arguments '%s' not found. "
```

Comment:
Use hanging indent:
```
msg = (
    "..."
)
```
Are the new messages tested? I think you could avoid the repetition of the two branches by interpolating "keyword " in the message and varying args/kwargs in the params with a variable.